src/main/java/seedu/address/model/slot/Slot.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.model.slot;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+import seedu.address.model.pet.Pet;
+
+/**
+ * Represents a Slot in the pet shop helper schedule system.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Slot implements Comparable<Slot> {
+
+    private final Pet pet;
+    private final LocalDateTime dateTime;
+    private final Duration duration;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Slot(Pet pet, LocalDateTime dateTime, Duration duration) {
+        requireAllNonNull(pet, dateTime, duration);
+        this.pet = pet;
+        this.dateTime = dateTime;
+        this.duration = duration;
+    }
+
+    public Pet getPet() {
+        return pet;
+    }
+
+    public LocalDateTime getDateTime() {
+        return dateTime;
+    }
+
+    public Duration getDuration() {
+        return duration;
+    }
+
+    public LocalDateTime getEndDateTime() {
+        //TODO
```

Comment:
I know this is a dummy but later can just dateTime.plus(duration), it works for my ip at least

Reply:
yep!